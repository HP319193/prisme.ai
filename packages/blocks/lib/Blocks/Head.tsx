import { useBlock } from '../Provider';
import { useBlocks } from '../Provider/blocksContext';

interface HeadConfig {
  content?: string;
}

export const HeadBlock = () => {
  const {
    config: { content = '' },
  } = useBlock<HeadConfig>();
  const {
    components: { Head },
  } = useBlocks();
  if (!Head) return null;
  return <Head>{content}</Head>;
};

export default HeadBlock;
