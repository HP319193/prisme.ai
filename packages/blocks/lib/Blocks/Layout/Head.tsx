import { useLayoutEffect, useMemo, useRef, useState } from 'react';
import { LayoutContext, useLayout } from './context';
import tw from '../../tw';
import { BlockLoader } from '../../BlockLoader';
import { BlockContext, useBlock } from '../../Provider';
import { LeftCircleOutlined } from '@ant-design/icons';

interface HeadRendererProps {
  blocks: LayoutContext['head'];
  onBack: () => void;
  hasHistory: boolean;
}
export const HeadRenderer = ({
  blocks,
  onBack,
  hasHistory,
  api,
  events,
}: HeadRendererProps & Pick<BlockContext, 'api' | 'events'>) => {
  const buttonEl = useRef<HTMLButtonElement>(null);
  const [animationClassName, setAnimationClassName] = useState('');

  const margin = buttonEl.current
    ? buttonEl.current.getBoundingClientRect().width * 2
    : 0;
  useLayoutEffect(() => {
    if (hasHistory) {
      setAnimationClassName('');
    } else {
      setAnimationClassName(tw`-translate-x-[${margin}px]`);
    }
  }, [hasHistory, margin]);

  return (
    <div className={tw`flex flex-1 bg-white transition-transform`}>
      <div
        className={`flex flex-row flex-1 transition-transform ${animationClassName}`}
        style={{ marginRight: `-${margin}px` }}
      >
        <button
          onClick={onBack}
          className={tw`head__button-back button-back ml-2`}
          ref={buttonEl}
        >
          <LeftCircleOutlined />
        </button>
        {blocks.map(({ block, ...config }, index) => (
          <BlockLoader
            key={index}
            config={config}
            name={block}
            api={api}
            events={events}
          />
        ))}
      </div>
    </div>
  );
};

export const Head = () => {
  const { back, history, head } = useLayout();
  const { api, events } = useBlock();
  return useMemo(
    () => (
      <HeadRenderer
        onBack={back}
        hasHistory={history.length > 1}
        blocks={head}
        api={api}
        events={events}
      />
    ),
    [back, history, api, events, head]
  );
};

export default Head;
