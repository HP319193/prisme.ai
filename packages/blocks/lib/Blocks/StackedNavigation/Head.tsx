import { useLayoutEffect, useMemo, useRef, useState } from 'react';
import { StackedNavigationContext, useStackedNavigation } from './context';
import tw from '../../tw';
import { LeftCircleOutlined } from '@ant-design/icons';
import { BlocksList } from '../BlocksList';
import { useBlocks } from '../../Provider/blocksContext';

interface HeadRendererProps {
  blocks: StackedNavigationContext['head'];
  onBack: () => void;
  hasHistory: boolean;
}
export const HeadRenderer = ({
  blocks,
  onBack,
  hasHistory,
}: HeadRendererProps) => {
  const {
    utils: { BlockLoader },
  } = useBlocks();
  const buttonEl = useRef<HTMLButtonElement>(null);
  const [animationClassName, setAnimationClassName] = useState('');

  const margin = buttonEl.current
    ? buttonEl.current.getBoundingClientRect().width * 2
    : 0;
  useLayoutEffect(() => {
    if (hasHistory) {
      setAnimationClassName('');
    } else {
      setAnimationClassName(tw`-translate-x-[${margin}px]`);
    }
  }, [hasHistory, margin]);

  const legacyBlocks = blocks.map(({ block, slug, ...rest }) => ({
    slug: slug || block || '',
    ...rest,
  }));

  return (
    <div className={tw`flex flex-1 bg-white transition-transform`}>
      <div
        className={`flex flex-row flex-1 transition-transform ${animationClassName}`}
        style={{ marginRight: `-${margin}px` }}
      >
        <button
          onClick={onBack}
          className={tw`head__button-back button-back ml-2`}
          ref={buttonEl}
        >
          <LeftCircleOutlined />
        </button>
        {blocks && (
          <BlockLoader
            name="BlocksList"
            config={{ blocks: legacyBlocks, className: 'flex-1' }}
          />
        )}
      </div>
    </div>
  );
};

export const Head = () => {
  const { back, history, head } = useStackedNavigation();

  return useMemo(
    () => (
      <HeadRenderer
        onBack={back}
        hasHistory={history.length > 1}
        blocks={head}
      />
    ),
    [back, history, head]
  );
};

export default Head;
