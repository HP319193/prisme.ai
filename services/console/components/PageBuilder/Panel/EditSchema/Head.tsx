import { Schema } from '@prisme.ai/design-system';

const schema: Schema = {
  type: 'object',
  properties: {
    content: {
      type: 'localized:string',
      title: 'pages.blocks.head.settings.content.label',
      description: 'pages.blocks.head.settings.content.description',
      'ui:widget': 'textarea',
    },
  },
};

export default schema;
