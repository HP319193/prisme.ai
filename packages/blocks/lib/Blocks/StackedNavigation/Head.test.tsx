import renderer from 'react-test-renderer';
import { HeadRenderer } from './Head';

jest.mock('../../Provider/blocksContext', () => {
  const mock = {
    utils: {
      BlockLoader: jest.fn(function BlockLoader({ onLoad }: any) {
        onLoad && onLoad();
        return null;
      }),
    },
  };
  const useBlocks = () => mock;
  return {
    useBlocks,
  };
});

it('should render with no block', () => {
  const blocks: Parameters<typeof HeadRenderer>[0]['blocks'] = [];
  const onBack = jest.fn();
  const root = renderer.create(
    <HeadRenderer blocks={blocks} onBack={onBack} hasHistory={false} />
  );
  expect(root.toJSON()).toMatchSnapshot();
});

it('should render with blocks', () => {
  const blocks: Parameters<typeof HeadRenderer>[0]['blocks'] = [
    {
      block: 'Header',
    },
  ];
  const onBack = jest.fn();
  const root = renderer.create(
    <HeadRenderer blocks={blocks} onBack={onBack} hasHistory={true} />
  );
  expect(root.toJSON()).toMatchSnapshot();
});
