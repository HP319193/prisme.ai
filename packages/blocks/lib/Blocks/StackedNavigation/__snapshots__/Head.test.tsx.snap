// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with blocks 1`] = `
<div
  className="flex flex-1 bg-white transition-transform"
>
  <div
    className="flex flex-row flex-1 transition-transform "
    style={
      Object {
        "marginRight": "-0px",
      }
    }
  >
    <button
      className="head__button-back button-back ml-2"
      onClick={[MockFunction]}
    />
    <div
      className="pr-block-blocks-list "
    />
  </div>
</div>
`;

exports[`should render with no block 1`] = `
<div
  className="flex flex-1 bg-white transition-transform"
>
  <div
    className="flex flex-row flex-1 transition-transform -translate-x-[0px]"
    style={
      Object {
        "marginRight": "-0px",
      }
    }
  >
    <button
      className="head__button-back button-back ml-2"
      onClick={[MockFunction]}
    />
    <div
      className="pr-block-blocks-list "
    />
  </div>
</div>
`;
